repa/grids/psdiffusion.hpp

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright 2017-2019 The repa authors
+ *
+ * This file is part of Repa.
+ *
+ * Repa is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Repa is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Repa.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "diffusion.hpp"
+
+namespace repa {
+namespace grids {
+
+struct PSDiffusion : public Diffusion {
```

Comment:
Please add a comment to this class. (Same as diffusion but preserves structure, etc./what does that mean?)

Reply:
Updated in 14f7a25